client/src/components/DashboardDropdown/DashboardDropdown.jsx

```diff
@@ -0,0 +1,94 @@
+import { React, useEffect, useState, useContext } from 'react';
+import { Link, useLocation, useNavigate } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { DarkModeContext } from '../../util/DarkModeProvider';
+import './DashboardDropdown.scss';
+import { userSelector } from '../../state/user/userSlice';
+
+const DashboardDropdown = ({ open, setOpen, items }) => {
+  const { darkMode, setDarkModeStatus } = useContext(DarkModeContext);
+  let count = 0; // counts which page
+
+  const { pathname } = useLocation();
+
+  return (
+    <>
+      <div
+        className={`dashboard-dropdown-bg ${
+          open ? 'dashboard-dropdown-bg-display' : 'dashboard-dropdown-bg-hide'
+        }`}
+        onClick={() => {
+          setOpen(false);
+        }}
+      >
+        <div
+          className={`dashboard-dropdown-container ${
+            open ? 'dashboard-dropdown-container-display' : ''
+          }`}
+        >
+          {items.map((item) => {
+            count++;
+            const { user } = useSelector(userSelector);
+            let hasAuthScope = false;
+            if (item.authScopes) {
+              for (let authScope of item.authScopes) {
+                if (user && user?.authScopes?.approved?.includes(authScope)) {
+                  hasAuthScope = true;
+                  break;
+                }
+              }
+            }
+
+            console.log('STATUS', hasAuthScope);
+
+            const hasAnyRegisterScope =
```

Comment:
instead of hasAnyRegisterScope, it would be best if we don't show the dropdown menus if the user has no scopes in that field (e.g., if user has no scope in scunt, don't show that dropdown), once again this does need some refactoring since it would be a lot easier if the dropdown click part and the dropdown image are in the same component.

we can come back to this later for enhancement unless you're not currently working on another task ;)